src/kirby/components/icon/custom-icon-name.directive.tns.ts

```diff
@@ -0,0 +1,65 @@
+import { LayoutBase } from 'tns-core-modules/ui/layouts/layout-base';
+import { Label } from 'tns-core-modules/ui/label/label';
+import {
+  Directive,
+  ElementRef,
+  Input,
+  Inject,
+  AfterViewInit,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+
+import { CUSTOM_FONT_SETTINGS, CustomIconSettings } from './custom-icon-settings';
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: '[customName]',
+})
+export class CustomIconNameDirective implements AfterViewInit, OnChanges {
+  // tslint:disable-next-line:no-input-rename
+  @Input('customName') customName: string;
+
+  constructor(
+    private element: ElementRef,
+    @Inject(CUSTOM_FONT_SETTINGS) private customIconSettings: CustomIconSettings
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.handleCustomIcon(this.customName);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.customName) {
+      this.handleCustomIcon(this.customName);
+    }
+  }
+
+  handleCustomIcon(name: string) {
+    if (name && this.customIconSettings[0].icons) {
+      const fontFamily = this.customIconSettings[0].fontfamily;
+      const icon = this.getCustomIcon(name);
+
+      if (fontFamily === undefined) {
+        console.warn('Missing font-family.');
```

Comment:
This is my personal opinion, but i find it better that the code blows up then warns me in a console, that i am using it wrong.

Point being, that if i remove the `fontFamily` or uses an `icon` that is not in the settings, there is something wrong and i may overlook a warning in the console.

Reply:
Agree - except, how do we catch this at compile time? Or will it be catched in tests?
If it only happens runtime we might not wanna fail hard?